src/libraries/System.IO.Compression.ZipFile/ref/System.IO.Compression.ZipFile.cs

```diff
@@ -11,13 +11,13 @@ public static partial class ZipFile
     {
         public static void CreateFromDirectory(string sourceDirectoryName, string destinationArchiveFileName) { }
         public static void CreateFromDirectory(string sourceDirectoryName, string destinationArchiveFileName, System.IO.Compression.CompressionLevel compressionLevel, bool includeBaseDirectory) { }
-        public static void CreateFromDirectory(string sourceDirectoryName, string destinationArchiveFileName, System.IO.Compression.CompressionLevel compressionLevel, bool includeBaseDirectory, System.Text.Encoding entryNameEncoding) { }
+        public static void CreateFromDirectory(string sourceDirectoryName, string destinationArchiveFileName, System.IO.Compression.CompressionLevel compressionLevel, bool includeBaseDirectory, System.Text.Encoding? entryNameEncoding) { }
         public static void ExtractToDirectory(string sourceArchiveFileName, string destinationDirectoryName) { }
         public static void ExtractToDirectory(string sourceArchiveFileName, string destinationDirectoryName, bool overwriteFiles) { }
         public static void ExtractToDirectory(string sourceArchiveFileName, string destinationDirectoryName, System.Text.Encoding entryNameEncoding) { }
```

Comment:
Should entryNameEncoding here be nullable?

Reply:
I believe it should. It's a perfectly valid use case [alluded to in the docs](https://github.com/dotnet/runtime/blob/c9d499d7e98df71ddc380600c2182db8773d6947/src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Create.cs#L330-L352). I assume you're asking because the above overload should cover that use case. However labeling as nullable is desirable for a couple of reasons:

1. It accurately reflects the contract of the particular methods.
2. It improves compositionality in certain scenaria: `(encoding != null) ? call_overload_1 : call_overload_2`

Reply:
> I assume you're asking because the above overload should cover that use case.

I'm asking because you didn't mark it as nullable but I think it should be.

Reply:
Ah apologies, I thought you were commenting on the change in line 14

Reply:
Fixed.